src/Uno.UI/UI/Xaml/Media/MatrixTransform.cs

```diff
@@ -18,7 +19,28 @@ public Matrix Matrix
 
 		public MatrixTransform()
 		{
+		}
+
+		protected override Rect TransformBoundsCore(Rect rect)
+		{
+			var leftTop = Matrix.Transform(new Point(rect.Left, rect.Top));
+			var rightTop = Matrix.Transform(new Point(rect.Right, rect.Top));
+			var rightBottom = Matrix.Transform(new Point(rect.Right, rect.Bottom));
+			var leftBottom = Matrix.Transform(new Point(rect.Left, rect.Bottom));
+
+			var x = Math.Min(Math.Min(leftTop.X, rightTop.X), Math.Min(rightBottom.X, leftBottom.X));
+			var y = Math.Min(Math.Min(leftTop.Y, rightTop.Y), Math.Min(rightBottom.Y, leftBottom.Y));
 
+			var width = Math.Max(Math.Max(leftTop.X, rightTop.X), Math.Max(rightBottom.X, leftBottom.X)) - x;
+			var height = Math.Max(Math.Max(leftTop.Y, rightTop.Y), Math.Max(rightBottom.Y, leftBottom.Y)) - y;
```

Comment:
FYI: there is an extension method `Rect.Transform(Matrix3x2)` which uses framework APIs for that, But as I already change this in my transforms branch, feel free to keep it like this :)